Mlem/App/Views/Root/Tabs/Inbox/InboxView.swift

```diff
@@ -10,46 +10,221 @@ import MlemMiddleware
+                Group {
+                    if showRefreshPopup {
+                        refreshPopup
+                            .transition(.move(edge: .bottom).combined(with: .opacity))
+                    }
+                }
+                .animation(.bouncy, value: showRefreshPopup)
+            }
+    }
+    
+    @ViewBuilder
+    var content: some View {
+        FancyScrollView {
+            LazyVStack(spacing: 0, pinnedViews: [.sectionHeaders]) {
+                Section {
+                    Divider()
+                    if loadingState == .loading, replies.isEmpty, mentions.isEmpty {
+                        ProgressView()
+                            .controlSize(.large)
+                            .padding()
+                            .tint(palette.secondary)
+                    } else {
+                        if items.isEmpty {
+                            Text("Nothing to see here")
+                                .foregroundStyle(palette.secondary)
+                                .padding()
+                        } else {
+                            ForEach(items, id: \.id) { item in
+                                VStack(alignment: .leading, spacing: 0) {
+                                    if let reply = item as? Reply2, !reply.creator.blocked {
+                                        ReplyView(reply: reply)
+                                    }
+                                    if let message = item as? Message2, !message.creator.blocked {
+                                        MessageView(message: message)
+                                    }
+                                    Divider()
+                                }
+                            }
+                        }
+                    }
+                } header: {
+                    BubblePicker(
+                        Tab.allCases,
+                        selected: $selectedTab,
+                        label: { tab in
+                            tab.rawValue.capitalized
+                        },
+                        value: { tab in
+                            if let unreadCount = (appState.firstSession as? UserSession)?.unreadCount {
+                                switch tab {
+                                case .all:
+                                    return unreadCount.total
+                                case .replies:
+                                    return unreadCount.replies
+                                case .mentions:
+                                    return unreadCount.mentions
+                                case .messages:
+                                    return unreadCount.messages
+                                }
+                            }
+                            return 0
+                        }
+                    )
+                    .background(palette.background.opacity(isAtTop ? 1 : 0))
+                    .background(.bar)
```

Comment:
*nit*

This is a pretty hefty view, worth moving to an `@ViewBuilder var header` helper